### PR TITLE
Add SEP-8 types and constants

### PR DESCRIPTION
**Pull request by howardtw**
**What**

This PR introduces SEP-8 types and constants. This PR also adds one extra field to the `PostActionUrlResponse` as SEP-8 has been updated.

**Next**

1. I'll be implementing the "approve" function in ApprovalProvider.
2. I'll be implementing the "postActionUrl" function in ApprovalProvider.
3. I'll be adding the helper functions to find the approval server URL.

### CODE REVIEW

File: src/types/sep8.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import { ApprovalResponseStatus } from "../constants/sep8";
+
+export interface ApprovalResponse {
+  status:
```
Comment (quietbits) [inline src/types/sep8.ts:4]: If any of the `ApprovalResponseStatus` value is accepted, we don't need to list all of them out. `status: ApprovalResponseStatus`.

File: src/types/sep8.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    | ApprovalResponseStatus.success
+    | ApprovalResponseStatus.revised
+    | ApprovalResponseStatus.pending
+    | ApprovalResponseStatus.actionRequired
+    | ApprovalResponseStatus.rejected;
+}
+
+export interface TransactionApproved extends ApprovalResponse {
```
Comment (quietbits) [inline src/types/sep8.ts:12]: I don't think we need to extend `ApprovalResponse` for these. There is only one attribute there at the moment and we set it for every extended type separately. We might keep them if there will be more shared attributes in the future.

**Review by quietbits — APPROVED** [review]

File: src/types/sep8.ts (hunk 2 above)
Comment (howardtw) [reply src/types/sep8.ts:12]: Got it. On the same topic, is it a common pattern to have the return type of a API call to be the base response, i.e. `ApprovalResponse` in this case, and depends on the `status`, we marshal the response to the associated extended type?

File: src/types/sep8.ts (hunk 1 above)
Comment (howardtw) [reply src/types/sep8.ts]: Looks much cleaner: https://github.com/stellar/js-stellar-wallets/pull/202/commits/64a48247d677483b2368dafe66662498d6c7e88d

**Pull request merged**